signal-compiler.c

```diff
@@ -38,53 +38,54 @@ int main(int ac, char *av[]) {
+      uint8_t sbuf[PGM_W] = {0};
+      uint8_t pgm_buf[PGM_W][PGM_H] = {0};
 
-  const int rv = better_read(sptr, sbuf, sizeof(sbuf));
-  (void)rv; // FIXME
+      const int rv = better_read(sptr, sbuf, sizeof(sbuf));
+      (void) rv; // FIXME
 
-  printf("read signal\n");
+      printf("read signal\n");
 
-  for (unsigned sx = 0; sx < PGM_W; sx++) {
-    pgm_buf[sx][sbuf[sx]] = PGM_DEPTH;
-  }
+      for (unsigned sx = 0; sx < PGM_W; sx++) {
+          pgm_buf[sx][sbuf[sx]] = PGM_DEPTH;
+      }
 
-  for (unsigned sx = 0; sx < PGM_W; sx++) {
-    const unsigned sy = sbuf[sx];
-    printf("%d %d\n", sx, sy);
+      for (unsigned sx = 0; sx < PGM_W; sx++) {
+          const unsigned sy = sbuf[sx];
+          // printf("%d %d\n", sx, sy);
 
-    // fill in gradient in square around sampled point
-    for (signed dx = -GRADIENT; dx <= GRADIENT; dx++) {
-      for (signed dy = -GRADIENT; dy <= GRADIENT; dy++) {
-        // {sx + dx, sy + dy}
-        if (0 <= (signed)sx + dx && (signed)sx + dx < PGM_W &&
-            0 <= (signed)sy + dy && (signed)sy + dy < PGM_H) {
-          uint8_t *p = &pgm_buf[sx + dx][sy + dy];
-          const uint8_t new = PGM_DEPTH - (PGM_DEPTH / GRADIENT / sqrt(2)) *
-                                              sqrt(pow(dx, 2) + pow(dy, 2));
-          if (*p == 0)
-            *p = new;
-          else
-            *p = *p > new ? *p : new;
-        }
+          // fill in gradient in square around sampled point
+          for (signed dx = -GRADIENT; dx <= GRADIENT; dx++) {
+              for (signed dy = -GRADIENT; dy <= GRADIENT; dy++) {
+                  // {sx + dx, sy + dy}
+                  if (0 <= (signed) sx + dx && (signed) sx + dx < PGM_W &&
+                      0 <= (signed) sy + dy && (signed) sy + dy < PGM_H) {
+                      uint8_t *p = &pgm_buf[sx + dx][sy + dy];
+                      const uint8_t new = PGM_DEPTH - (PGM_DEPTH / GRADIENT / sqrt(2)) *
+                                                      sqrt(pow(dx, 2) + pow(dy, 2));
+                      if (*p == 0)
+                          *p = new;
+                      else
+                          *p = *p > new ? *p : new;
+                  }
+              }
+          }
       }
-    }
-  }
 
-  // - write(pptr, pgm_buf, PGM_H * PGM_W)
-  // - write(pptr, pgm_buf[ii], PGM_H)
-  // - write(pptr, &pgm_buf[ii][jj], 1)   <-- use this one (byte-order reasons)
-  for (unsigned jj = 0; jj < PGM_H; jj++) {
-    for (unsigned ii = 0; ii < PGM_W; ii++) {
-      write(pptr, &pgm_buf[ii][jj], 1);
-    }
+      // - write(pptr, pgm_buf, PGM_H * PGM_W)
+      // - write(pptr, pgm_buf[ii], PGM_H)
+      // - write(pptr, &pgm_buf[ii][jj], 1)   <-- use this one (byte-order reasons)
+      for (unsigned jj = 0; jj < PGM_H; jj++) {
+          for (unsigned ii = 0; ii < PGM_W; ii++) {
+              write(pptr, &pgm_buf[ii][jj], 1);
+          }
+      }
+      printf("wrote %ui to pgm\n", sizeof(pgm_buf));
```

Comment:
Also I changed this because the make wouldn't let it pass.

Reply:
oh weird! i think `%ld` was what `clang` gave a suggestion for. good to know